packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -391,7 +391,7 @@ Future<Null> diagnoseXcodeBuildFailure(
       // Make sure the user has specified one of:
       // DEVELOPMENT_TEAM (automatic signing)
       // PROVISIONING_PROFILE (manual signing)
-      !(app.buildSettings?.containsKey('DEVELOPMENT_TEAM')) == true
+      !app.buildSettings?.containsKey('DEVELOPMENT_TEAM') == true
           || app.buildSettings?.containsKey('PROVISIONING_PROFILE') == true) {
```

Comment:
This if statement is really suspect.

1. there's a `!` on nullable
2. the indentation seems to show that there should be parenthesis around the `||`

wdyt ?

btw the line 401 looks also strange. I wonder if there shouldn't be parenthesis around `app.id?.contains('com.example') ?? false` .

Reply:
yeah this is super suspect, i was just thinking that to myself before I saw your comment!

cc @cbracken @xster

Reply:
Yikes. Nice catch. I'll take a quick look.

Reply:
Ah man, thanks for the catch. It's totally wrong. It should be 
```dart
!(app.buildSettings?.containsKey('DEVELOPMENT_TEAM') == true
    || app.buildSettings?.containsKey('PROVISIONING_PROFILE') == true)) {
```

Reply:
401 should have parenthesis like you're saying too.

Reply:
#14478 and #14480